core/src/main/scala/pink/cozydev/lucille/Query.scala

```diff
@@ -27,6 +27,16 @@ sealed trait Query extends Product with Serializable {
     * @return
     */
   def mapLastTerm(f: Query.Term => Query): Query
+
+  def and(q: Query): Query = Query.And(this, q)
+
+  def or(q: Query): Query = Query.Or(this, q)
+
+  def not: Query = Query.Not(this)
+
+  def group(q: Query): Query = Query.Group(this, q)
```

Comment:
I can't quite remember how these get formed, but a group could consist of more than two queries. 

Would it be worth allowing for a pattern like `query.group(query2, query3, query4)`?? (I think the signature would end up looking basically the same as the existing apply, `def group(q: Query, maybeOtherQueries: Query*)`)

Reply:
Oh wow, thank you for bringing this up!
I think you've highlighted something that isn't going to work well with "Problem 2" as called out in https://github.com/cozydev-pink/lucille/issues/142

We should actually not have a `group` helper method that takes in another query at all!
```scala
def group(q: Query): Query = Query.Group(this, q)
```
This leaves us in the situation where we do not have a clear boolean operator. Are we suppose to AND `this` and `q` or `OR` them together?
I think we should either drop the `group` helper method for now and introduce it once we the "Problem 2" situation is addressed, or we should change it to be a simple wrapping operation:
```scala
def group(q: Query): Query = Query.Group(this)
```

However, this admittedly feels a bit strange given that you can still manually construct a `Group` with multiple queries (without a clear boolean operation).

@astronomerdamo maybe we just drop `group` for now?

Reply:
Hiii Sam, M'Day!

Fair point - I've dropped the group helper dot method. Apologies this took so long.